apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -212,9 +348,59 @@ export const ModelList = ({
             disabled={disabled}
             searchTerm={searchTerm}
             isReplayAsIs={isReplayAsIs}
+            openApplicationModal={() => setModalIsOpen(true)}
+            setIsDeleteModalOpen={setIsDeleteModalOpen}
+            setCurrentEntityName={setCurrentEntityName}
+            setCurrentEntityId={setCurrentEntityId}
+            setCurrentEntityReference={setCurrentEntityReference}
+            handlePublish={handlePublish}
           />
         ))}
       </div>
+      {isDeleteModalOpen && (
+        <ConfirmDialog
+          isOpen={isDeleteModalOpen}
+          heading="Confirm deleting application"
+          description="Are you sure you want to delete the application?"
+          confirmLabel="Delete"
+          cancelLabel="Cancel"
+          onClose={(result) => {
+            setIsDeleteModalOpen(false);
+            if (result) {
+              handleDelete();
+            }
+          }}
+        />
+      )}
+      {modalIsOpen && (
+        <ApplicationDialog
+          isOpen={modalIsOpen}
+          onClose={() => setModalIsOpen(false)}
+          mode="edit"
+          selectedApplication={applicationDetail}
+          currentReference={currentEntityReference}
+        />
+      )}
+      {applicationDetail && (
+        <PublishModal
+          entity={{
+            name: applicationDetail.display_name,
+            id: ApiUtils.decodeApiUrl(
+              applicationDetail.name ||
+                (applicationDetail as unknown as { application: string })
+                  .application,
+            ),
+            folderId: getFolderIdFromEntityId(applicationDetail.display_name),
+          }}
```

Comment:
we shouldn't collect entity here. It's not readable code and should be moved out of markup into variable